### ISSUE TITLE
Fourier transform from real to complex space has no adjoint

### ISSUE THREAD

**Issue by adler-j** [odlgroup/odl#389]
This error pops up any time you attempt to use the Fourier transform in an inverse problem, basically rendering it useless except in pre-post processing. This is a shame considering how cool it is :*(

The adjoint exists in _some_ sense mathematically. What steps do we need to take to make this workable?

**Comment by kohr-h** [comment]
Okay, I thought about this for a second. Here is how we can save the Fourier transform:

Strictly speaking, there is no adjoint for a real-to-complex transform. However, we can identify the range with the corresponding product space of real spaces and define
```
F: L^2(R^d, R) --> [L^2(R^d, R)]^2
F(f) = (F_c(f), -F_s(f))
```
where `F_c` and `F_s` are the cosine and sine transforms, respectively. The adjoint of this transform is well-defined and given as
```
F^*: [L^2(R^d, R)]^2 --> L^2(R^d, R)
F*(g) = F_c(g_1) - F_s(g_2)
```
since both `F_c` and `F_s` are self-adjoint.

If we compare this to the naive approach of taking the real part of the "usual" adjoint, one gets
```
F^*(g) = Re[ F^*(Re(g)) + i * F^*(Im(g)) ] = F_c(Re(g)) - F_s(Im(g))
```
which is the same as above.

So, in conclusion, the naive approach is all fine. Perhaps the cast to real could be explicit in the code. Also, adding a short section in the doc about this identification would not hurt.

**Comment by adler-j** [comment]
This theory agrees with my simple experiment of simply removing the "is real" check (which gave me working examples).  Agree with conclusion.
